routers/api/v1/repo/issue.go

```diff
@@ -902,3 +903,399 @@ func UpdateIssueDeadline(ctx *context.APIContext) {
+	// - name: index
+	//   in: path
+	//   description: index of the issue
+	//   type: string
+	//   required: true
+	// - name: body
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/IssueMeta"
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/Issue"
+	//   "404":
+	//     description: the issue does not exist
+
+	createDep(ctx, models.DependencyTypeBlocking)
+}
+
+// RemoveIssueBlocking unblock the issue given in the body by the issue in path
+func RemoveIssueBlocking(ctx *context.APIContext) {
+	// swagger:operation DELETE /repos/{owner}/{repo}/issues/{index}/blocks issue issueRemoveIssueBlocking
+	// ---
+	// summary: Unblock the issue given in the body by the issue in path
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the issue
+	//   type: string
+	//   required: true
+	// - name: body
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/IssueMeta"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Issue"
+
+	removeDep(ctx, models.DependencyTypeBlocking)
+}
+
+func createDep(ctx *context.APIContext, t models.DependencyType) {
+	if !ctx.Repo.Repository.IsDependenciesEnabled() {
+		ctx.NotFound()
+		return
+	}
+
+	dep, err := models.GetIssueWithAttrsByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound("IsErrIssueNotExist", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	form := web.GetForm(ctx).(*api.IssueMeta)
+	repo, err := repo_model.GetRepositoryByOwnerAndName(form.Owner, form.Name)
+	if err != nil {
+		if repo_model.IsErrRepoNotExist(err) {
+			ctx.NotFound("IsErrRepoNotExist", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetRepositoryByOwnerAndName", err)
+		}
+		return
+	}
+
+	issue, err := models.GetIssueWithAttrsByIndex(repo.ID, form.Index)
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound("IsErrIssueNotExist", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	if t == models.DependencyTypeBlockedBy {
+		err = models.CreateIssueDependency(ctx.User, issue, dep)
+	} else {
+		err = models.CreateIssueDependency(ctx.User, dep, issue)
+	}
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "CreateIssueDependency", err)
+		return
+	}
+
+	ctx.JSON(http.StatusCreated, convert.ToAPIIssue(dep))
+}
+
+func removeDep(ctx *context.APIContext, t models.DependencyType) {
+	if !ctx.Repo.Repository.IsDependenciesEnabled() {
+		ctx.NotFound()
+		return
+	}
+
+	issue, err := models.GetIssueWithAttrsByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
```

Comment:
Again we're not rendering this issue so we don't need all of its comments and attachments.